app/models/custom_attribute.rb

```diff
@@ -2,7 +2,17 @@ class CustomAttribute < ApplicationRecord
   belongs_to :resource, :polymorphic => true
   serialize :serialized_value
 
+  def value=(value)
+    self.serialized_value = value
+    self[:value] = value
+  end
+
   def stored_on_provider?
     source == "VC"
   end
+
+  def field_type
+    field_type = serialized_value.class
+    field_type == NilClass ? :string : field_type.to_s.downcase.to_sym
```

Comment:
There are more then that, we can take care of all of these:
https://github.com/alongoldboim/manageiq/blob/360dbe841ce2548e3720c8d495e7ae8d5b974716/lib/miq_expression.rb#L1364